test/FunctionalTests/Infrastructure/HttpResponseMessageExtensions.cs

```diff
@@ -47,17 +48,30 @@ public static async Task<T> GetSuccessfulGrpcMessageAsync<T>(this HttpResponseMe
 
         public static void AssertTrailerStatus(this HttpResponseMessage response, StatusCode statusCode, string details)
         {
-            var trailerValueString = response.TrailingHeaders.GetValues(GrpcProtocolConstants.StatusTrailer).Single();
-            Assert.AreEqual(statusCode.ToTrailerString(), trailerValueString, $"Expected grpc-status {statusCode} but got {(StatusCode)Convert.ToInt32(trailerValueString)}");
+            var statusString = GetStatusValue(response.TrailingHeaders, GrpcProtocolConstants.StatusTrailer)
```

Comment:
same as in previous PR: 
if the status or messages are not found in trailers, this always checks the headers without checking the preconditions if for RPC response the "Trailers-Only" case is admissible.